optax/contrib/_ano.py

```diff
@@ -0,0 +1,178 @@
+    def _update_v(g_t, v_prev):
+      g2 = jnp.square(g_t).astype(v_prev.dtype)
+      b2_t = jnp.asarray(b2, dtype=v_prev.dtype)
+      one_minus_b2_t = jnp.asarray(1.0 - b2, dtype=v_prev.dtype)
+      sign_term = jnp.sign(g2 - v_prev)
+      return b2_t * v_prev + one_minus_b2_t * sign_term * g2
+
+    nu = jax.tree.map(_update_v, g, state.nu)
+
+    # Bias correction for second moment (scalar), cast per-leaf at use-site.
+    bias_correction2_scalar = (
+      1.0 - jnp.asarray(b2, dtype=jnp.float32) ** count_inc
+    )
+
+    # Direction: |g| * sign(m) / sqrt(v_hat + eps), all in leaf dtype.
+    def _direction(g_t, m_t, v_t):
+      bc2 = jnp.asarray(bias_correction2_scalar, dtype=v_t.dtype)
+      v_hat = v_t / bc2
+      eps_t = jnp.asarray(eps, dtype=v_t.dtype)
+      denom = jnp.sqrt(v_hat + eps_t)
+      sgn = jnp.sign(m_t).astype(g_t.dtype)
+      return jnp.abs(g_t) * sgn / denom
+
+    direction = jax.tree.map(_direction, g, mu, nu)
+    mu = optax.tree.cast(mu, mu_dtype)
+    return direction, transform.ScaleByAdamState(count=count_inc, mu=mu, nu=nu)
+
+  return base.GradientTransformation(init_fn, update_fn)
+
+
+def ano(
+    learning_rate: base.ScalarOrSchedule,
+    b1: float = 0.92,
+    b2: float = 0.99,
+    eps: float = 1e-8,
+    weight_decay: float = 0.0,
+    logarithmic_schedule: bool = False,
+    mu_dtype: Optional[Any] = None,
+) -> base.GradientTransformationExtraArgs:
+  r"""ANO optimizer.
+
+  ANO uses sign–magnitude decoupling (sign of momentum for direction, gradient
+  magnitude for scaling) with an additive (Yogi-like) second-moment update.
+
+  Args:
+    learning_rate: A global scaling factor, either fixed or evolving along
+      iterations with a scheduler.
+    b1: First-moment decay β1.
+    b2: Parameter for second-moment update β2.
+    eps: Small constant ε added inside the square root.
+    weight_decay: Decoupled weight decay coefficient.
+    logarithmic_schedule: If True, use logarithmic
+      schedule for b1: 1-1/log(max(2,k)).
+    mu_dtype: Optional dtype for the first order accumulator m.
+
+  Returns:
+    The corresponding :class:`optax.GradientTransformationExtraArgs`.
+
+  Examples:
+    >>> import optax
+    >>> import jax
+    >>> import jax.numpy as jnp
+    >>> def f(x): return jnp.sum(x ** 2)
+    >>> solver = optax.contrib.ano(learning_rate=0.003)
+    >>> params = jnp.array([1., 2., 3.])
+    >>> opt_state = solver.init(params)
+    >>> for _ in range(5):
+    ...  grad = jax.grad(f)(params)
+    ...  updates, opt_state = solver.update(grad, opt_state, params)
+    ...  params = optax.apply_updates(params, updates)
+    ...  print('Objective function: {:.2E}'.format(f(params)))
+    Objective function: 1.40E+01
+    Objective function: 1.39E+01
+    Objective function: 1.39E+01
+    Objective function: 1.39E+01
+    Objective function: 1.38E+01
+
+  References:
+    Kegreisz, `Ano: Faster is Better in Noisy Landscapes
+    <https://github.com/Adrienkgz/ano-optimizer>`_.
```

Comment:
Maybe would be nice to link the paper here instead of the repo (or alongside it)?